src/w3d/math/vector4.cpp

```diff
@@ -0,0 +1,23 @@
+/**
+ * @file
+ *
+ * @author Tiberian Technologies
+ * @author OmniBlade
+ *
+ * @brief 4D Vector class.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "vector4.h"
+
+#include "vector3.h"
+
```

Comment:
You could add a note that these must hold true for `To_Vector3_Ptr` to work

Reply:
Good point. I added hint.